core/domain/user_services.py

```diff
@@ -1355,8 +1355,16 @@ def update_profile_picture_data_url(
         profile_picture_data_url: str. New profile picture url to be set.
     """
     user_settings = get_user_settings(user_id, strict=True)
-    user_settings.profile_picture_data_url = profile_picture_data_url
-    save_user_settings(user_settings)
+    username = user_settings.username
+    fs = fs_services.GcsFileSystem(feconf.ENTITY_TYPE_USER, username)
+    filename_png = 'profile_picture.png'
+    fs.commit(filename_png, profile_picture_data_url, mimetype='image/png')
+
+    profile_picture_webp = webptools.base64str2webp_base64str(
+        base64str=profile_picture_data_url, image_type="png",
+        option="-q 80",logging="-v")
+    filename_webp = 'profile_picture.webp'
+    fs.commit(filename_webp, profile_picture_webp, mimetype='image/webp')
```

Comment:
Should we still work with the file as a data URL? Shouldn't we work with bytes already?

Reply:
I have updated this part.